text/0097-unbonding_queue.md

```diff
@@ -0,0 +1,166 @@
+
+The combination of long unbonding periods and high returns has lead to the proliferation of [liquid staking](https://www.bitcoinsuisse.com/learn/what-is-liquid-staking), where parachains or centralised exchanges offer users their staked tokens before the 28 days unbonding period is over either in original DOT/KSM form or derivative tokens. Liquid staking is harmless if few tokens are involved but it could result in many validators being selected by a few entities if a large fraction of DOTs were involved. This may lead to centralization (see [here](https://dexola.medium.com/is-ethereum-about-to-get-crushed-by-liquid-staking-30652df9ec46) for more discussion on threats of liquid staking) and an opportunity for attacks.  
+
+The new mechanism greatly increases the competitiveness of Polkadot, while maintaining sufficient security.
+
+
+## Stakeholders
+
+- Every DOT/KSM token holder
+
+## Explanation
+
+Before diving into the details of how to implement the unbonding queue, we give readers context about why Polkadot has a 28-day unbonding period in the first place. The reason for it is to prevent long-range attacks (LRA) that becomes theoretically possible if more than 1/3 of validators collude. In essence, a LRA describes the inability of users, who disconnect from the consensus at time t0 and reconnects later, to realize that validators which were legitimate at a certain time, say t0 but dropped out in the meantime, are not to be trusted anymore. That means, for example, a user syncing the state could be fooled by trusting validators that fell outside the active set of validators after t0, and are building a competitive and malicious chain (fork). 
+
+LRAs of longer than 28 days are mitigated by the use of trusted checkpoints, which are assumed to be no more than 28 days old. A new node that syncs Polkadot will start at the checkpoint and look for proofs of finality of later blocks, signed by 2/3 of the validators. In an LRA fork, some of the validator sets may be different but only if 2/3 of some validator set in the last 28 days signed something incorrect. 
+
+If we detect an LRA of no more than 28 days with the current unbonding period, then we should be able to detect misbehaviour from over 1/3 of validators whose nominators are still bonded. The stake backing these validators is considerable fraction of the total stake (empirically it is 0.287 or so). If we allowed more than this stake to unbond, without checking who it was backing, then the LRA attack might be free of cost for an attacker. The proposed mechansim allows up to half this stake to unbond within 28 days. This halves the amount of tokens that can be slashed, but this is still very high in absolute terms. For example, at the time of writing (19.06.2024) this would translate to around 120 millions DOTs.
+
+Attacks other than an LRA, such as backing incorrect parachain blocks, should be detected and slashed within 2 days. This is why the mechanism has a minimum unbonding period.
```

Comment:
Ok, I am not to familiar with staking internals, but I will assume that if you unbond, your stake will always remain at least for the duration of the current era. Hypothetically on a network with eras longer than 2 days, it could happen that your stake is securing the network until the end of the era and gets unbonded the very next block. Thus, causing an offense at the end of an era would be risk free?

If that is true, we should bind the unbonding period to multiples of an era, e.g.  2, instead of a fixed 2 days. Also it would be good to measure that period in number of blocks, because then a potential DoS attack would be pointless as it would also increase the unbonding time.

Reply:
The 2 days here is used to make it easier for the reader and putting durations into a more familiar context. The queue itself, however, is specified in blocks in the chapter "Mechanism". Here, `MAX_DURATION` is equal to 403200 blocks and `MIN_DURATION` is equal to 28800. If the queue is implemented on other networks with different era durations, the parameters can easily be adjusted.

Reply:
Perfect on using blocks! For the relationship to eras: Would be good if that relationship to era duration would be at least documented (not necessarily in this RFC as it is already up for vote), but for sure in the implementation.

Reply:
Makes sense, I'll make to add it to the documentation for the implementation.